### ISSUE TITLE
Potential bug with array searchable attributes

### ISSUE THREAD

**Issue by Fronix** [meilisearch/meilisearch#4244]
**Describe the bug**
[Original discord thread](https://discord.com/channels/1006923006964154428/1180169287374225418)
Was helping a person on how to make an array of object properties searchable and encountered an odd _potential_ bug. Given a simple set of objects that includes a `translations` array with a `title` the text in the `title` can be searched if configured correctly in `searchableAttributes` but the word `Ability` gives no matches. All other words seem to be searchable.

**To Reproduce**
1. Download [message.json](https://github.com/meilisearch/meilisearch/files/13536064/message.json)
2. Create a new index with default settings
3. Add `translations.title` as the only searchable attribute
4. Searching for `to accept criticism` gives matches
5. Searching for `criticism`, gives matches
6. Searching for `Ability`, **gives no matches**

**Expected behavior**
Simply put, I'm expecting `Ability` to match several documents.

**Screenshots**
![image](https://github.com/meilisearch/meilisearch/assets/631180/943950fe-f42b-43a0-bc69-61398d6114e5)
![image](https://github.com/meilisearch/meilisearch/assets/631180/ce9347d8-b4a4-4860-bd87-0b172f06051b)
![image](https://github.com/meilisearch/meilisearch/assets/631180/3f98ce26-0e51-44ed-9582-2f96a4dfb2bb)

**Meilisearch version:**
v1.5.0

Additional information that may be relevant to the issue.
Runs in docker

**Comment by irevoire** [comment]
I reproduced the bug on the latest main.
It's really strange because no results are returned with "Ability" when this document exists:
```json
{
      "id": "d81983f7-f7dc-4ddd-8fec-4e6cc39a7bd0",
      "translations": [
        {
          "id": "064ee085-cab1-44f4-8908-3ff52c1f677b",
          "languageCode": "EN",
          "title": "Ability to keep deadlines"
        },
        {
          "id": "ec40ab8f-ec31-4248-af78-aa9490f0f6e7",
          "languageCode": "HY",
          "title": "Ժամկետները պահպանելու ունակություն"
        }
      ],
      "type": "SOFT",
      "title": "Ability to keep deadlines"
    }
```

But I do get results when I look for any other word _even when they're at the start of the document_ like "Constructive":
```json
{
      "id": "eaa4e19d-adf4-49bf-adc3-1d98d6ed7e13",
      "translations": [
        {
          "id": "dff49658-5449-4c26-899a-8c0f3891560c",
          "languageCode": "EN",
          "title": "Constructive feedback giving"
        },
        {
          "id": "95cbebfb-0b4b-4c3e-9cb4-4c94867c125e",
          "languageCode": "HY",
          "title": "Կառուցողական հետադարձ կապի տրամադրում"
        }
      ],
      "type": "SOFT",
      "title": "Constructive feedback giving"
    }
```

The only difference I can see between `Ability` and any other word is that there are 8 documents starting with `Ability` without only one or two for every other word.

**Comment by jzabroski** [comment]
If you point me to the area of code that likely does this, happy to spend time and debug it.

**Comment by NarHakobyan** [comment]
any news? :)

**Comment by NarHakobyan** [comment]
Sorting also works incorrectly

https://github.com/meilisearch/meilisearch/assets/1781506/c128fa72-8a24-499a-b704-f394fb191d43

**Comment by irevoire** [comment]
Hey @jzabroski,

> Sorting also works incorrectly

Which version are you using?
We fixed a bug in the latest version of meilisearch (v1.6.2)

**Comment by irevoire** [comment]
Hey @NarHakobyan,

After updating to the latest version, have you re-indexed your documents?
Since the bug corrupts the database, you need to re-index everything to see the fix.

The easiest way to do that if you can’t clear your index and send back all your documents is to:
- Remove all your filterable attributes
- **Wait until the task is fully processed**; this is very important; otherwise, there is a risk meilisearch understand what you’re doing and optimize it out
- Send back all your filterable attributes

If you updated your version through a dump, none of this applies.

**Comment by curquiza** [comment]
Hello @Fronix 

Do you still have the issue with [v1.10.0](https://github.com/meilisearch/meilisearch/releases/tag/v1.10.0)?

**Comment by Fronix** [comment]
Currently not able to test but maybe https://github.com/NarHakobyan, who is the user I was helping with the issue can test?

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by jzabroski** [comment]
> After updating to the latest version, have you re-indexed your documents?

I suspect this is the key and why I initially offered my help to test this.  I think the README should be clearer about refreshing indices when ugprading meilisearch.  This is something I have discussed / hinted at in the past but would like to see movement on as I think it would help your core team members a lot.  I am not a big brain, just somebody who likes quick wins.

**Comment by curquiza** [comment]
@jzabroski It's not common to have to reindex everything after upgrading Meilisearch. It's due to a specific bug, and we try to make clear in the related release notes. Sorry for the inconvenience.

Let us know @NarHakobyan
